parlai/mturk/core/dev/react_server/dev/components/socket_handler.jsx

```diff
@@ -632,18 +574,20 @@ class SocketHandler extends React.Component {
       assignment_id: this.props.assignment_id,
       sender_id: this.props.worker_id,
       conversation_id: this.props.conversation_id,
-      type: TYPE_HEARTBEAT,
+      agent_status: this.props.agent_status,
+      done_text: this.props.done_text,
       data: null,
+      received_message_count: used_message_ids.size,
     };
 
-    this.safePacketSend({ type: SOCKET_ROUTE_PACKET_STRING, content: hb });
+    this.safePacketSend({ type: HEARTBEAT, content: hb });
 
     this.setState({
-      heartbeats_without_pong: this.state.heartbeats_without_pong + 1,
+      heartbeats_without_response: this.state.heartbeats_without_response + 1,
     });
-    if (this.state.heartbeats_without_pong >= 12) {
+    if (this.state.heartbeats_without_response >= 12) {
```

Comment:
what are these magic numbers?

Reply:
**Magic**. Actually they've just been selected somewhat arbitrarily. I should move them to constants and put a TODO to actually iterate and determine good values.